src/radical/saga/adaptors/slurm/slurm_job.py

```diff
@@ -619,6 +619,18 @@ def _job_run(self, jd):
             # use '-C EGRESS' to enable outbound network
             script += "#SBATCH -C EGRESS\n"
 
+        elif 'comet' in self.rm.host.lower():
+
+            if gpu_count:
+                # gres resources are specified *per node*
+                assert(n_nodes), 'need unique number of cores per node'
+
+                if gpu_arch: gpu_arch = gpu_arch.lower()
+                else       : gpu_arch = 'p100'
+                count = 4
+                # Make sure we take a full GPU node
+                script += "#SBATCH --gres=gpu:%s:%d\n" % (gpu_arch, count)
```

Comment:
proposing the following changes
```python
                    gres_list = ['gpu']
                    if gpu_arch: gres_list.append(gpu_arch.lower())
                    gres_list.append(str(count))
                    script += "#SBATCH --gres=%s\n" % ':'.join(gres_list)
```
to consider the option when `gpu_arch` is not set

Reply:
Wouldn't that change semantics?  At the moment, `p100` is set when `gpu_arch` is unset.  Also, I don't think the changed code is easier to read.

Reply:
> At the moment, p100 is set when gpu_arch is unset

right, but it is possible to let the system pick the gpu, e.g., here what they provide in the documentation

> #SBATCH --gres=gpu:4        #first available gpu node (either type)
> #SBATCH --gres=gpu:k80:4    #only K80 nodes
> #SBATCH --gres=gpu:p100:4   #only P100 nodes

p.s. and yeah, code could be changed

Reply:
> let the system pick the gpu

I see, makes sense.   @lee212 : would that work for you?

Reply:
Yes, thanks for the changes